SofaKernel/framework/sofa/helper/vector.cpp

```diff
@@ -51,6 +52,62 @@ void SOFA_HELPER_API vector_access_failure(const void* vec, unsigned size, unsig
     assert(i < size);
 }
 
-} // namespace helper
+/// Convert the string 's' into an unsigned int. The error are reported in msg & numErrors
+/// is incremented.
+int SOFA_HELPER_API getInteger(const std::string& s, std::stringstream& msg, unsigned int& numErrors)
+{
+    const char* attrstr=s.c_str();
+    char* end=nullptr;
+    int retval = strtol(attrstr, &end, 10);
+
+    /// It is important to check that the string was totally parsed to report
+    /// message to users because a silent error is the worse thing that can happen in UX.
+    if(end !=  attrstr+strlen(attrstr))
+    {
+        if(numErrors<5)
+            msg << "    - problem while parsing '" << s <<"' as Integer'. Replaced by 0 instead." << msgendl ;
+        if(numErrors==5)
+            msg << "   - ... " << msgendl;
+        numErrors++ ;
+        return 0 ;
+    }
+    return retval ;
+}
+
+/// Convert the string 's' into an unsigned int. The error are reported in msg & numErrors
+/// is incremented.
+unsigned int SOFA_HELPER_API getUnsignedInteger(const std::string& s, std::stringstream& msg, unsigned int& numErrors)
+{
+    const char* attrstr=s.c_str();
+    char* end=nullptr;
+
+    /// If there is minus sign we exit.
+    if(s.find_first_of('-') != std::string::npos ){
```

Comment:
most likely you want to use a seekable `std::istream` instead and operate directly on the file stream